### PR TITLE
Implement TRS changes including aliases and search by ID

### PR DESCRIPTION
**Pull request by denis-yuen**
* adds testing coverage for TRS
* addresses search by ID https://github.com/ga4gh/dockstore/issues/1486
* adds search by alias and ability to add aliases in the first place using Dockstore APIs  #1395

Note: this PR uses JPA queries rather than native SQL. Will migrate the rest of the native code since it doesn't understand polymorphism.

### CODE REVIEW

**Comment by denis-yuen** [comment]
Note to self: this requires a TRS update

**Comment by denis-yuen** [comment]
(@garyluu  ignore that, was an experiment)

File: dockstore-webservice/src/main/resources/openapi.yaml [hunk 1]
```diff
@@ -825,6 +837,12 @@ paths:
           required: false
           schema:
             type: string
+        - name: checker
+          in: query
+          description: Return only checker workflows
+          required: false
```
Comment (garyluu) [inline dockstore-webservice/src/main/resources/openapi.yaml]: set default false

File: dockstore-webservice/src/main/resources/openapi.yaml [hunk 2]
```diff
@@ -4843,6 +4903,11 @@ components:
       name: Authorization
       in: header
   schemas:
+    Alias:
+      type: object
+      properties:
+        content:
+          type: string
```
Comment (garyluu) [inline dockstore-webservice/src/main/resources/openapi.yaml]: Is there a plan on adding more properties to this?

File: dockstore-webservice/pom.xml [hunk 3]
```diff
@@ -839,7 +844,7 @@
                 <groupId>io.swagger</groupId>
                 <artifactId>swagger-codegen-maven-plugin</artifactId>
                 <configuration>
-                    <inputSpec>https://raw.githubusercontent.com/ga4gh/tool-registry-service-schemas/2.0.0-beta.1/src/main/resources/swagger/ga4gh-tool-discovery.yaml</inputSpec>
+                    <inputSpec>https://raw.githubusercontent.com/ga4gh/tool-registry-service-schemas/feature/search/src/main/resources/swagger/ga4gh-tool-discovery.yaml</inputSpec>
```
[the remark this answers is not in the document]
Comment (denis-yuen) [reply dockstore-webservice/pom.xml]: Yeah, we're in a weird place where I need a TRS update for this, I think we're basically going to need a TRS unstable branch

File: dockstore-webservice/src/main/resources/openapi.yaml (hunk 1 above)
Comment (denis-yuen) [reply dockstore-webservice/src/main/resources/openapi.yaml]: (this is generated)

File: dockstore-webservice/src/main/resources/openapi.yaml (hunk 2 above)
Comment (denis-yuen) [reply dockstore-webservice/src/main/resources/openapi.yaml]: I believe so

File: dockstore-webservice/src/main/resources/openapi.yaml (hunk 1 above)
Comment (garyluu) [reply dockstore-webservice/src/main/resources/openapi.yaml]: ya, i was thinking to set it as false in the `@ApiParam(..., defaultValue=false)`

File: dockstore-webservice/src/main/resources/swagger.yaml [hunk 4]
```diff
@@ -3332,7 +3383,7 @@ paths:
         schema:
           type: "array"
           items:
-            $ref: "#/definitions/SourceFile"
+            type: "object"
```
[the remark this answers is not in the document]
Comment (agduncan94) [reply dockstore-webservice/src/main/resources/swagger.yaml]: :(

File: dockstore-webservice/src/main/resources/openapi.yaml (hunk 1 above)
Comment (denis-yuen) [reply dockstore-webservice/src/main/resources/openapi.yaml]: I think the API declaration is generated too (TRS swagger -> server API)

File: dockstore-webservice/src/main/resources/swagger.yaml (hunk 4 above)
[the remark this answers is not in the document]
Comment (denis-yuen) [reply dockstore-webservice/src/main/resources/swagger.yaml]: FYI, I might need to burn some time and tackle this once and for all. 
It actually seems to be deterministically generating the wrong value now.

**Pull request merged**